lib/dtutils.lua

```diff
@@ -0,0 +1,457 @@
+      pat - string - the pattern to split on]],
+  Description = [[split separates a string into a table of strings.  The strings are separated at each
+    occurrence of the supplied pattern.]],
+  Return_Value = [[result - a table of strings on success, or an empty table on error]],
+  Example = [[split("/a/long/path/name/to/a/file.txt", "/") would return a table like
+      {"a", "long", "path", "name", "to", "a", "file.txt"}]],
+}
+
+-- Thanks to http://lua-users.org/wiki/SplitJoin
+function dtutils.split(str, pat)
+   local t = {}  -- NOTE: use {n = 0} in Lua-5.0
+   local fpat = "(.-)" .. pat
+   local last_end = 1
+   local s, e, cap = str:find(fpat, 1)
+   while s do
+      if s ~= 1 or cap ~= "" then
+        table.insert(t,cap)
+      end
+      last_end = e+1
+      s, e, cap = str:find(fpat, last_end)
+   end
+   if last_end <= #str then
+      cap = str:sub(last_end)
+      table.insert(t, cap)
+   end
+   return t
+end
+
+--[[
+  NAME
+    join - join a table of strings with a specified separator
+
+  SYNOPSIS
+    local du = require "lib/dtutils"
+
+    local result = du.join(tabl, pat)
+      tabl - a table of strings
+      pat - a separator
+
+  DESCRIPTION
+    join assembles a table of strings into a string with the specified pattern 
+    in between each string
+
+  RETURN VALUE
+    result - the joined string on success, or an empty string on failure
+
+  EXAMPLE
+    join({"a", "long", "path", "name", "to", "a", "file.txt"}, " ") would return the string
+      "a long path name to a file.txt"
+
+]]
+
+dtutils.libdoc.functions[#dtutils.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value", "Example"},
+  Name = [[join - join a table of strings with a specified separator]],
+  Synopsis = [[local du = require "lib/dtutils"
+
+    local result = du.join(tabl, pat)
+      tabl - a table of strings
+      pat - a separator]],
+  Description = [[join assembles a table of strings into a string with the specified pattern 
+    in between each string]],
+  Return_Value = [[result - the joined string on success, or an empty string on failure]],
+  Example = [[join({"a", "long", "path", "name", "to", "a", "file.txt"}, " ") would return the string
+      "a long path name to a file.txt"]],
+}
+
+-- Thanks to http://lua-users.org/wiki/SplitJoin
+function dtutils.join(tabl, pat)
+  returnstr = ""
+  for i,str in pairs(tabl) do
+    returnstr = returnstr .. str .. pat
+  end
+  return string.sub(returnstr, 1, -(pat:len() + 1))
+end
+
+--[[
+  NAME
+    prequire - a protected lua require
+
+  SYNOPSIS
+    local du = require "lib/dtutils"
+
+    local result = du.prequire(req_name)
+      req_name - the filename of the lua code to load without the ".lua" filetype
+
+  DESCRIPTION
+    prequire is a protected require that can survive an error in the code being loaded without
+    bringing down the calling routine.
+
+  RETURN VALUE
+    result - the code or true on success, otherwise an error message
```

Comment:
That's fine.  I use this in script_manager and plugin_manager so that a "bad" script doesn't bring everything crashing down.